app/lib/shared/handler_helpers.dart

```diff
@@ -76,6 +77,50 @@ shelf.Handler wrapHandler(
   return handler;
 }
 
+/// Detects simple redirect loop and emits a log message on it.
+shelf.Handler _redirectLoopDetectorWrapper(
+    Logger logger, shelf.Handler handler) {
+  return (shelf.Request request) async {
+    final rs = await handler(request);
+    if (rs.statusCode >= 300 && rs.statusCode < 400) {
+      final location = rs.headers[HttpHeaders.locationHeader];
+
+      // sanity check for the header being present
+      if (location == null) {
+        logger.shout('Redirect response without location header.',
+            Exception('Redirect response without location header.'));
+        return rs;
+      }
+
+      // sanity check for the header being valid
+      final uri = Uri.tryParse(location);
+      if (uri == null) {
+        logger.shout(
+            'Redirect response with invalid location header: "$location".',
+            Exception(
+                'Redirect response with invalid location header: "$location".'));
+        return rs;
+      }
+
+      // exact match
+      if (request.requestedUri == uri) {
+        logger.shout(
+            'Redirect loop detected.', Exception('Redirect loop detected.'));
+        return rs;
+      }
+
+      // path + querystring match
+      if (uri.toString().startsWith(uri.path) &&
+          request.requestedUri.toString().endsWith(uri.toString())) {
+        logger.shout(
+            'Redirect loop detected.', Exception('Redirect loop detected.'));
+        return rs;
+      }
```

Comment:
This is if you do `pub.dev/path -> /path` ?

Reply:
Yes, also matching it with `?queryString`.